tools/gn_test_wrapper/lib/gn_test_wrapper.dart

```diff
@@ -0,0 +1,178 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:async/async.dart';
+import 'package:collection/collection.dart';
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+
+import 'src/run_wrapper_internal.dart';
+
+/// Entrypoint from the command line to run a test executable produced by GN.
+///
+/// ## Example
+///
+/// ```dart
+/// void main(List<String> args) async => runWrapper(args, MyGnTestWrapper());
+/// ```
+Future<void> runWrapper(
+  List<String> args,
+  GnTestWrapper wrapper, {
+  StringSink? stdout,
+  StringSink? stderr,
+}) async {
+  await runWrapperInternal(args, wrapper, stdout: stdout, stderr: stderr);
+}
+
+/// Wraps and standardizes the invocation of a test executable produced by GN.
+///
+/// A [GnTestWrapper] is a stateless immutable object that represents an
```

Comment:
This would be a good place to plumb standard options from `et test` that we require all test wrappers to support (e.g. --test-filter=)